Sources/Sentry/Profiling/MemorySafety.mm

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Specto Inc. All rights reserved.
+
+// From https://github.com/kstenerud/KSCrash/blob/master/Source/KSCrash/Recording/Tools/KSMemory.c
+//
+//  Copyright (c) 2012 Karl Stenerud. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall remain in place
+// in this source code.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <mach/mach.h>
+
+namespace specto::darwin {
```

Comment:
`m`: We already have this code in the repo https://github.com/getsentry/sentry-cocoa/blob/master/Sources/SentryCrash/Recording/Tools/SentryCrashMemory.c. Can we find a way that this is not duplicated?

Reply:
The APIs do seem interchangeable, but it will not compile if I include that file into ours:
<img width="1363" alt="image" src="https://user-images.githubusercontent.com/3241469/151637508-a28dc932-0fb4-44b6-9c78-f24aeb0d9273.png">

Any guidance on how I could resolve this?